pkg/connection/store/kubernetes/store.go

```diff
@@ -93,7 +93,7 @@ func buildClient(ctx context.Context, local client.Client, cfg v1.SecretStoreCon
 // ReadKeyValues reads and returns key value pairs for a given Kubernetes Secret.
 func (ss *SecretStore) ReadKeyValues(ctx context.Context, n store.ScopedName, s *store.Secret) error {
 	ks := &corev1.Secret{}
-	if err := ss.client.Get(ctx, types.NamespacedName{Name: n.Name, Namespace: ss.namespaceForSecret(n)}, ks); err != nil {
+	if err := ss.client.Get(ctx, types.NamespacedName{Name: n.Name, Namespace: ss.namespaceForSecret(n)}, ks); resource.IgnoreNotFound(err) != nil {
```

Comment:
if not found, then what would be the value of `ks.Data`?

could we add a unit test for the new behavior?

If ESS is not enabled, then should we return the error?

Reply:
> > It will be an empty map.
> 
> Can we update the functions docs to reflect that?

It shouldn't be necessary. In real world, we shouldn't end up with an empty secret. This change is done to proceed with an intermediate stage where the claim/composite secret is not created yet but it's fetched by the PTF [composer](https://github.com/crossplane/crossplane/blob/2df2202d4ddc846bfdcf7f1cce7bc82c55d28010/internal/controller/apiextensions/composite/composition_ptf.go#L286) to observe the state. Later in the reconciliation loop, the secret will be created as usual and we don't need to handle not found errors.

Reply:
or is the caller relying on the fact that Metadata is set even on not found?

Reply:
I think if we cannot find the secret on the store, we should overwrite the given secret fields with empty/nil values.